### PR TITLE
Add support for installing Oracle JDK, JRE 6, 7 and 8 on RedHat.

### PR DESCRIPTION
**Pull request by mmarseglia**
Hello,

  We needed support for Oracle Java and I added support for it here.  I have tested this on CentOS and it seems to work OK.

  Could you please take a look?  It would really help if Puppet Labs officially supported Oracle Java.

Thank you

### CODE REVIEW

**Comment by deiga** [comment]
This is a pretty crucial addition, please support this!

**Comment by WhatsARanjit** [comment]
@mmarseglia You might consider swapping in the nanliu/staging module instead for fetching files.  Also, edit the metadata.json to add a dependency to a new module.

**Comment by mmarseglia** [comment]
@WhatsARanjit added archive and updated metadata.json

**Comment by tphoney** [comment]
Thanks for the work, i have a few initial comments, the commits should be squashed and you should probably not commit the log file.

**Comment by mmarseglia** [comment]
@tphoney no problem, is this better?

**Comment by hunner** [comment]
@mmarseglia It looks like something went wrong. You have 8 commits in this PR (instead of 1) and 7 of them are from other people.

If you reset your master branch back to the latest upstream master branch and then cherry pick `dd32b28` onto it and force push it should clear it up. Something like `git checkout master && git fetch --all && git reset --hard puppetlabs/master` (or whatever the puppetlabs remote is called) to reset your branch, then `git cherry-pick dd32b28 && git push --force origin master` to push just your commit to the PR.

In the future, I would suggest making PRs from branches named something other than `master` because it can be confusing having a different master branch than upstream.

**Comment by mmarseglia** [comment]
@hunner Oh no, sorry about that!  I'll look at this problem and fix it.  Thank you for the guidance.

File: manifests/oracle.pp [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  # full URL to download
+  $url = "${baseUrl}${releaseMajor}-${releaseMinor}/${packageName}"
+
+  case $ensure {
+    'present' : {
+      archive { $destination :
+        ensure  => present,
+        source  => $url,
+        cleanup => false,
+        cookie  => 'gpw_e24=http%3A%2F%2Fwww.oracle.com%2F; oraclelicense=accept-securebackup-cookie',
```
Comment (hunner) [inline manifests/oracle.pp]: This is circumventing the license agreement, yeah? So... not something we should advocate.

**Comment by mmarseglia** [comment]
I'm not a lawyer but I would say you're agreeing to it no matter how you download it.  Could put a warning to that effect in the README.

**Comment by mmarseglia** [comment]
@hunner since this is a puppetlabs supported module, I would appreciate it if Puppetlabs could figure out whether or not it is acceptable to download the Oracle JDK/JRE by setting the cookie.

I prefer not to go through the trouble of rebasing if this isn't going to be merged based on the licensing issues.

Thanks,

**Comment by hunner** [comment]
@mmarseglia You may not be a lawyer but I was just hoping you'd link to a legal reason for why it is legal. Anyway, on http://www.oracle.com/technetwork/java/javase/terms/license/index.html it says "[...] BY USING THE SOFTWARE YOU ACKNOWLEDGE THAT YOU HAVE READ THE TERMS AND AGREE TO THEM" so that sounds good to me. Wanna rebase?

File: manifests/oracle.pp (hunk 1 above)
Comment (hunner) [inline manifests/oracle.pp]: Oracle BCL says it's okay, actually.

Comment (mmarseglia) [inline manifests/oracle.pp]: Thanks, I'll rebase.

**Comment by hunner** [comment]
Adding archive dependency is backwards incompatible, so this will need a major bump.

**Comment by tphoney** [comment]
@mmarseglia we will discuss this on Thursday at the community meeting

**Comment by bmjen** [comment]
@mmarseglia Thanks for the effort. We haven't heard back in a while on your progress with this PR. Due to the inactivity as well as the fact that this is backwards-incompatible. I'm going to close this PR for now.

File: metadata.json [hunk 2]
```diff
@@ -70,14 +78,15 @@
...
     {
       "name": "puppet",
-      "version_requirement": "3.x"
+      "version_requirement": ">= 3.0.0 < 5.0.0"
     }
   ],
   "dependencies": [
-    {"name":"puppetlabs/stdlib","version_requirement":">= 2.4.0"}
+    {"name":"puppetlabs/stdlib","version_requirement":">= 2.4.0 < 5.0.0"},
+    {"name":"puppet-community/puppet-archive","version_requirement":"0.3.0"}
```
Comment (hunner) [inline metadata.json]: This is now voxpupuli/puppet-archive

**Pull request closed without merging**